multipoint.go

```diff
@@ -40,14 +73,55 @@ func (g *MultiPoint) MustSetCoords(coords []Coord) *MultiPoint {
 	return g
 }
 
+// Coord returns the ith coord of g.
+func (g *MultiPoint) Coord(i int) Coord {
+	before := 0
+	if i > 0 {
+		before = g.ends[i-1]
+	}
+	if g.ends[i] == before {
+		return nil
+	}
+	return g.flatCoords[before:g.ends[i]]
+}
+
 // SetCoords sets the coordinates.
 func (g *MultiPoint) SetCoords(coords []Coord) (*MultiPoint, error) {
-	if err := g.setCoords(coords); err != nil {
-		return nil, err
+	g.flatCoords = nil
+	g.ends = nil
+	for _, c := range coords {
+		if len(c) > 0 {
```

Comment:
Here we implicitly have two representations for an empty point: either `nil` or `[]float64{}` (a `Coord` with no elements). I think we should only use `nil` and `[]float64{}` should return a `ErrLayoutMismatch` error instead.

Reply:
i buy that argument! done.